src/components/widgets/CategoriesSidebar.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="categories-sidebar">
+    <el-button class="category--all" type="primary" @click="changeCategory(null)">Все объявления</el-button>
+    <hr class="color-hovered-button" />
+    <el-button
+      v-for="category in categories"
+      :key="category.id"
+      :class="{ 'category--selected': current == category.id }"
+      class="category"
+      type="primary"
+      @click="changeCategory(category.id)"
+    >
+      {{ category.title }}
+    </el-button>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { ApiAdvertisementCategoryType } from '@/api/KY/AdvertisementService/advertisement.types';
+import { onBeforeMount, ref } from 'vue';
+import { advertisementApi } from '@/api/KY/AdvertisementService/advertisement.api';
+
+const props = defineProps<{
+  modelValue: number | null;
+}>();
+
+const emits = defineEmits<{
+  (e: 'update:modelValue', value: number | null): void;
+}>();
+
+const categories = ref<Array<ApiAdvertisementCategoryType>>([]);
+
+const current = ref(props.modelValue);
+
+const changeCategory = async (value: number | null) => {
+  current.value = value;
+
+  emits('update:modelValue', value);
+};
+
+onBeforeMount(async () => {
+  const [error, data] = await advertisementApi.getAllCategories();
+
+  console.log(error);
+
+  if (!error && data) {
+    categories.value = data;
+  }
+});
+</script>
+
+<style lang="scss" scoped>
+.categories-sidebar {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  border-radius: 5px;
+  background-color: $color--box;
+  padding: 15px;
+  row-gap: 10px;
+}
+
+.el-button.category {
```

Comment:
У тебя класс поменяется уже. Но чтобы ты знал, вложения делаются подобным образом, если у компонента два класса
```suggestion
.el-button {
  &.cateogry {}
 }
```